test/goalc/framework/test_runner.cpp

```diff
@@ -0,0 +1,117 @@
+
+#include "test_runner.h"
+
+#include <string>
+
+#include "gtest/gtest.h"
+#include "third-party/inja.hpp"
+#include "third-party/json.hpp"
+
+#include "game/runtime.h"
+#include "goalc/listener/Listener.h"
+#include "goalc/compiler/Compiler.h"
+
+#include "common/util/FileUtil.h"
+#include <filesystem>
+
+namespace GoalTest {
+
+std::string escaped_string(const std::string& in) {
+  std::string result;
+  for (auto x : in) {
+    switch (x) {
+      case '\n':
+        result.append("\\n");
+        break;
+      case '\t':
+        result.append("\\t");
+        break;
+      default:
+        result.push_back(x);
+    }
+  }
+  return result;
+}
+
+void CompilerTestRunner::run_static_test(inja::Environment& env,
+                                         std::string& testCategory,
+                                         const std::string& test_file,
+                                         const std::vector<std::string>& expected,
+                                         MatchParam<int> truncate) {
```

Comment:
Don't worry about changing this now, but C++17 now has `std::optional` which is better than the MatchParam thing.  Eventually I'll switch everything to std::optional